components/scream/src/physics/rrtmgp/zenith.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef ZENITH_HPP
+#define ZENITH_HPP
+
+/*
+ * Okay, things we do in EAM:
+ * get_cosine_solar_zenith -> zenith -> shr_orb_decl -> shr_orb_avg_cosz
+ * But we need orbital parameters from shr_orb_params in shr_orb_mod.
+ * It looks to me like we could start by compiling the fortran source
+ * shr_orb_mod, and then providing C++ bridges to shr_orb_decl,
+ * shr_orb_avg_cosz, and shr_orb_params. We could then have a zenith
+ * function that would call, in order:
+ *
+ *     shr_orb_params()
+ *     shr_orb_decl()
+ *     shr_orb_avg_cosz()
+ *
+ * That should at least get us a baseline, and then we can work on porting
+ * shr_orb_mod.F90 to C++.
+ */
+
+extern void shr_orb_params(double calday, double eccen, double mvelpp, double lambm0,
```

Comment:
The keyword `extern` along is somewhat pointless, since functions already have external linkage by default.

Reply:
By the way, I do not see any definition for `shr_orb_params`. I do see a `bind(C,name='shr_orb_params_c')` in a f90 module, but there's a name mismatch. Also, assuming that `shr_orb_params_c` is the function definition corresponding to dhis function declaration, you should declare this function as `extern "C"`, not just `extern`. The "C" part is crucial, since it tells the linker to link this function using C name mangling, even if this file is included in a cpp file compiler with c++ compilers.

Reply:
This was a temporary file as I was figuring out how to implement this that I mistakenly left behind after deciding to just use the fortran codes for the time being. This file has since been removed. `shr_orb_params` is in the top level share source, called from the bridged defined in this PR. I do have `extern “C”` defined for these. Please do verify that I’ve taken care of this properly.